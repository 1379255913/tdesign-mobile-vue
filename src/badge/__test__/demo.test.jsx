/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import basicVue from '@/badge/demos/basic.vue';
import colorVue from '@/badge/demos/color.vue';
import distVue from '@/badge/demos/dist.vue';
import maxCountVue from '@/badge/demos/max-count.vue';
import mobileVue from '@/badge/demos/mobile.vue';
import offsetVue from '@/badge/demos/offset.vue';
import shapeVue from '@/badge/demos/shape.vue';
import textVue from '@/badge/demos/text.vue';

const mapper = {
  basicVue,
  colorVue,
  distVue,
  maxCountVue,
  mobileVue,
  offsetVue,
  shapeVue,
  textVue,
};

describe('Badge', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Badge ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
