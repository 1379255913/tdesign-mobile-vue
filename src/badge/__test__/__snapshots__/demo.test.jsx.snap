// Vitest Snapshot v1

exports[`Badge > Badge basicVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
>
  <div
    class="t-cell-group"
  >
    <!--v-if-->
    <div
      class="t-cell-group__container"
    >
      
      <div
        class="badge-demo"
      >
        <div
          class="badge-item"
        >
          <div
            class="t-badge"
          >
            <div
              class="t-badge__inner t-badge--has-children t-badge--medium t-badge--circle"
            >
              16
            </div>
            
            消息
            
          </div>
        </div>
        <div
          class="badge-item"
        >
          <div
            class="t-badge"
          >
            <div
              class="t-badge__inner t-badge--has-children t-badge--medium t-badge--dot"
            >
              
            </div>
            
            消息
            
          </div>
        </div>
        <div
          class="badge-item"
        >
          <div
            class="t-badge"
          >
            <!--v-if-->
            
            消息
            
          </div>
        </div>
        <div
          class="badge-item"
        >
          <div
            class="t-badge"
          >
            <!--v-if-->
            
            消息
            
          </div>
        </div>
      </div>
      
    </div>
  </div>
</div>
`;

exports[`Badge > Badge colorVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
>
  <div
    class="t-cell-group"
  >
    <!--v-if-->
    <div
      class="t-cell-group__container"
    >
      
      <div
        class="badge-demo"
      >
        <div
          class="badge-item"
        >
          <div
            class="t-badge badge-item"
          >
            <div
              class="t-badge__inner t-badge--has-children t-badge--medium t-badge--circle"
            >
              12
            </div>
            
            <button
              aria-disabled="false"
              class="t-button t-size-s t-button--base t-button--default t-button--shape-rectangle"
              role="button"
            >
              <!---->
              <span
                class="t-button__text"
              >
                
                按钮
                
              </span>
            </button>
            
          </div>
          <div
            class="t-badge badge-item"
          >
            <div
              class="t-badge__inner t-badge--has-children t-badge--medium t-badge--circle"
              style="background: rgb(82, 196, 26);"
            >
              99+
            </div>
            
            <button
              aria-disabled="false"
              class="t-button t-size-s t-button--base t-button--default t-button--shape-rectangle"
              role="button"
            >
              <!---->
              <span
                class="t-button__text"
              >
                
                绿色
                
              </span>
            </button>
            
          </div>
          <div
            class="t-badge badge-item"
          >
            <div
              class="t-badge__inner t-badge--has-children t-badge--medium t-badge--circle"
              style="background: rgb(245, 47, 255);"
            >
              1
            </div>
            
            <button
              aria-disabled="false"
              class="t-button t-size-s t-button--base t-button--default t-button--shape-rectangle"
              role="button"
            >
              <!---->
              <span
                class="t-button__text"
              >
                
                自定义颜色
                
              </span>
            </button>
            
          </div>
        </div>
      </div>
      
    </div>
  </div>
</div>
`;

exports[`Badge > Badge distVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
  data-v-03ebeeaa=""
>
  <div
    class="badge-wrap"
    data-v-03ebeeaa=""
  >
    <div
      class="t-badge badge-item"
      data-v-03ebeeaa=""
    >
      <div
        class="t-badge__inner t-badge--has-children t-badge--medium t-badge--dot"
      >
        
      </div>
      
      <button
        aria-disabled="false"
        class="t-button t-size-s t-button--base t-button--default t-button--shape-rectangle"
        data-v-03ebeeaa=""
        role="button"
      >
        <!---->
        <span
          class="t-button__text"
        >
          
          按钮
          
        </span>
      </button>
      
    </div>
  </div>
  <div
    class="badge-wrap"
    data-v-03ebeeaa=""
  >
    <div
      class="t-cell-group"
      data-v-03ebeeaa=""
    >
      <!--v-if-->
      <div
        class="t-cell-group__container"
      >
        
        <div
          class="t-cell t-cell--middle t-cell--bordered"
          data-v-03ebeeaa=""
          value-align="left"
        >
          <div
            class="t-cell__left-icon"
          >
            <!--v-if-->
            <!--v-if-->
          </div>
          <!--v-if-->
          <div
            class="t-cell__note"
          >
            
            <span
              data-v-03ebeeaa=""
            >
              单行标题
            </span>
            <div
              class="t-badge list-item-badge"
              data-v-03ebeeaa=""
            >
              <div
                class="t-badge__inner t-badge--medium t-badge--dot"
                style="right: 10px; top: 0px;"
              >
                
              </div>
              <!---->
            </div>
            
          </div>
          <!--v-if-->
        </div>
        
      </div>
    </div>
  </div>
</div>
`;

exports[`Badge > Badge maxCountVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
>
  <div
    class="badge-demo"
  >
    <div
      class="t-badge badge-item"
    >
      <div
        class="t-badge__inner t-badge--has-children t-badge--medium t-badge--circle"
      >
        12
      </div>
      
      <button
        aria-disabled="false"
        class="t-button t-size-s t-button--base t-button--default t-button--shape-rectangle"
        role="button"
      >
        <!---->
        <span
          class="t-button__text"
        >
          
          14
          
        </span>
      </button>
      
    </div>
    <div
      class="t-badge badge-item"
    >
      <div
        class="t-badge__inner t-badge--has-children t-badge--medium t-badge--circle"
      >
        99+
      </div>
      
      <button
        aria-disabled="false"
        class="t-button t-size-s t-button--base t-button--default t-button--shape-rectangle"
        role="button"
      >
        <!---->
        <span
          class="t-button__text"
        >
          
          默认max:99
          
        </span>
      </button>
      
    </div>
    <div
      class="t-badge badge-item"
    >
      <div
        class="t-badge__inner t-badge--has-children t-badge--medium t-badge--circle"
      >
        44+
      </div>
      
      <button
        aria-disabled="false"
        class="t-button t-size-s t-button--base t-button--default t-button--shape-rectangle"
        role="button"
      >
        <!---->
        <span
          class="t-button__text"
        >
          
          max:44
          
        </span>
      </button>
      
    </div>
    <div
      class="t-badge badge-item"
    >
      <div
        class="t-badge__inner t-badge--has-children t-badge--medium t-badge--circle"
      >
        105+
      </div>
      
      <button
        aria-disabled="false"
        class="t-button t-size-s t-button--base t-button--default t-button--shape-rectangle"
        role="button"
      >
        <!---->
        <span
          class="t-button__text"
        >
          
          max:105
          
        </span>
      </button>
      
    </div>
  </div>
  <div
    class="badge-demo"
  >
    <div
      class="t-cell-group"
    >
      <!--v-if-->
      <div
        class="t-cell-group__container"
      >
        
        <div
          class="t-cell t-cell--middle t-cell--bordered"
          value-align="left"
        >
          <div
            class="t-cell__left-icon"
          >
            <!--v-if-->
            <!--v-if-->
          </div>
          <!--v-if-->
          <div
            class="t-cell__note"
          >
            
            <span>
              单行标题
            </span>
            <div
              class="t-badge list-item-badge"
            >
              <div
                class="t-badge__inner t-badge--medium t-badge--circle"
                style="right: 10px; top: 0px;"
              >
                16
              </div>
              <!---->
            </div>
            
          </div>
          <!--v-if-->
        </div>
        
      </div>
    </div>
  </div>
</div>
`;

exports[`Badge > Badge mobileVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
>
  <h1
    class="title"
  >
    Badge 徽标
  </h1>
  <p
    class="summary"
  >
    用于告知用户，该区域的状态变化或者待处理任务的数量。
  </p>
  <div
    class="tdesign-mobile-demo-block"
  >
    <h2
      class="tdesign-mobile-demo-block__title"
    >
      01 类型
    </h2>
    <p
      class="tdesign-mobile-demo-block__summary"
    >
      徽标主要分红点、数字、文字和角标提醒
    </p>
    
    <div
      class="badge-demo"
    >
      <div
        class="badge-item"
      >
        <div
          class="t-badge"
        >
          <div
            class="t-badge__inner t-badge--has-children t-badge--medium t-badge--dot"
          >
            
          </div>
          
          消息
          
        </div>
      </div>
      <div
        class="badge-item"
      >
        <div
          class="t-badge"
        >
          <div
            class="t-badge__inner t-badge--has-children t-badge--medium t-badge--circle"
          >
            16
          </div>
          
          消息
          
        </div>
      </div>
      <div
        class="badge-item"
      >
        <div
          class="t-badge"
        >
          <div
            class="t-badge__inner t-badge--has-children t-badge--medium t-badge--circle"
          >
            NEW
          </div>
          
          消息
          
        </div>
      </div>
      <div
        class="badge-item"
      >
        <div
          class="t-badge"
        >
          <div
            class="t-badge__inner t-badge--has-children t-badge--medium t-badge--circle"
          >
            ···
          </div>
          
          消息
          
        </div>
      </div>
    </div>
    <div
      class="badge-demo"
    >
      <div
        class="badge-item"
      >
        <div
          class="t-badge"
        >
          <div
            class="t-badge__inner t-badge--has-children t-badge--medium t-badge--dot"
          >
            
          </div>
          
          <button
            aria-disabled="false"
            class="t-button t-size-s t-button--outline t-button--default t-button--shape-rectangle"
            role="button"
          >
            <!---->
            <span
              class="t-button__text"
            >
              
              小按钮
              
            </span>
          </button>
          
        </div>
      </div>
      <div
        class="badge-item"
      >
        <div
          class="t-badge"
        >
          <div
            class="t-badge__inner t-badge--has-children t-badge--medium t-badge--circle"
          >
            16
          </div>
          
          <button
            aria-disabled="false"
            class="t-button t-size-s t-button--outline t-button--default t-button--shape-rectangle"
            role="button"
          >
            <!---->
            <span
              class="t-button__text"
            >
              
              小按钮
              
            </span>
          </button>
          
        </div>
      </div>
      <div
        class="badge-item"
      >
        <div
          class="t-badge"
        >
          <div
            class="t-badge__inner t-badge--has-children t-badge--medium t-badge--circle"
          >
            NEW
          </div>
          
          <button
            aria-disabled="false"
            class="t-button t-size-s t-button--outline t-button--default t-button--shape-rectangle"
            role="button"
          >
            <!---->
            <span
              class="t-button__text"
            >
              
              小按钮
              
            </span>
          </button>
          
        </div>
      </div>
      <div
        class="badge-item"
      >
        <div
          class="t-badge"
        >
          <div
            class="t-badge__inner t-badge--has-children t-badge--medium t-badge--circle"
          >
            ···
          </div>
          
          <button
            aria-disabled="false"
            class="t-button t-size-s t-button--outline t-button--default t-button--shape-rectangle"
            role="button"
          >
            <!---->
            <span
              class="t-button__text"
            >
              
              小按钮
              
            </span>
          </button>
          
        </div>
      </div>
    </div>
    <div
      class="t-cell-group"
      style="overflow: hidden;"
    >
      <!--v-if-->
      <div
        class="t-cell-group__container"
      >
        
        <div
          class="t-cell t-cell--middle t-cell--bordered"
        >
          <div
            class="t-cell__left-icon"
          >
            <!--v-if-->
            <!--v-if-->
          </div>
          <div
            class="t-cell__title"
          >
            单行标题
            <!--v-if-->
            <!--v-if-->
          </div>
          <div
            class="t-cell__note"
          >
            
            <div
              class="t-badge"
            >
              <div
                class="t-badge__inner t-badge--medium t-badge--dot"
              >
                
              </div>
              <!---->
            </div>
            
          </div>
          <div
            class="t-cell__right-icon"
          >
            <svg
              class="t-icon t-icon-chevron-right"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
        </div>
        <div
          class="t-cell t-cell--middle t-cell--bordered"
        >
          <div
            class="t-cell__left-icon"
          >
            <!--v-if-->
            <!--v-if-->
          </div>
          <div
            class="t-cell__title"
          >
            单行标题
            <!--v-if-->
            <!--v-if-->
          </div>
          <div
            class="t-cell__note"
          >
            
            <div
              class="t-badge"
            >
              <div
                class="t-badge__inner t-badge--medium t-badge--circle"
              >
                16
              </div>
              <!---->
            </div>
            
          </div>
          <div
            class="t-cell__right-icon"
          >
            <svg
              class="t-icon t-icon-chevron-right"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
        </div>
        <div
          class="t-cell t-cell--middle t-cell--bordered"
        >
          <div
            class="t-cell__left-icon"
          >
            <!--v-if-->
            <!--v-if-->
          </div>
          <div
            class="t-cell__title"
          >
            单行标题
            <!--v-if-->
            <!--v-if-->
          </div>
          <div
            class="t-cell__note"
          >
            
            <div
              class="t-badge"
            >
              <div
                class="t-badge__inner t-badge--medium t-badge--round"
              >
                NEW
              </div>
              <!---->
            </div>
            <div
              class="t-badge"
              style="margin-left: 8px;"
            >
              <div
                class="t-badge__inner t-badge--medium t-badge--circle"
              >
                NEW
              </div>
              <!---->
            </div>
            
          </div>
          <div
            class="t-cell__right-icon"
          >
            <svg
              class="t-icon t-icon-chevron-right"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M6.46 12.46l-.92-.92L9.08 8 5.54 4.46l.92-.92L10.92 8l-4.46 4.46z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
        </div>
        <div
          class="t-cell t-cell--middle t-cell--bordered"
        >
          <div
            class="t-cell__left-icon"
          >
            <!--v-if-->
            <!--v-if-->
          </div>
          <div
            class="t-cell__title"
          >
            单行标题
            <!--v-if-->
            <!--v-if-->
          </div>
          <div
            class="t-cell__note"
          >
            
            <div
              class="t-badge t-badge__ribbon--outer"
            >
              <div
                class="t-badge__inner t-badge--medium t-badge--ribbon"
              >
                NEW
              </div>
              <!---->
            </div>
            
          </div>
          <!--v-if-->
        </div>
        
      </div>
    </div>
    
  </div>
</div>
`;

exports[`Badge > Badge offsetVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
>
  <div
    class="badge-demo"
  >
    <div
      class="badge-line"
    >
      <span>
        offset: [10, 10]
      </span>
      <div
        class="t-badge badge-item"
      >
        <div
          class="t-badge__inner t-badge--has-children t-badge--medium t-badge--circle"
          style="right: 10px; top: 10px;"
        >
          12
        </div>
        
        <button
          aria-disabled="false"
          class="t-button t-size-s t-button--base t-button--default t-button--shape-rectangle"
          role="button"
        >
          <!---->
          <span
            class="t-button__text"
          >
            
            按钮
            
          </span>
        </button>
        
      </div>
    </div>
    <div
      class="badge-line"
    >
      <span>
        offset: [-10, 10]
      </span>
      <div
        class="t-badge badge-item"
      >
        <div
          class="t-badge__inner t-badge--has-children t-badge--medium t-badge--circle"
          style="right: -10px; top: 10px;"
        >
          12
        </div>
        
        <button
          aria-disabled="false"
          class="t-button t-size-s t-button--base t-button--default t-button--shape-rectangle"
          role="button"
        >
          <!---->
          <span
            class="t-button__text"
          >
            
            按钮
            
          </span>
        </button>
        
      </div>
    </div>
    <div
      class="badge-line"
    >
      <span>
        offset: [-10, -10]
      </span>
      <div
        class="t-badge badge-item"
      >
        <div
          class="t-badge__inner t-badge--has-children t-badge--medium t-badge--circle"
          style="right: -10px; top: -10px;"
        >
          12
        </div>
        
        <button
          aria-disabled="false"
          class="t-button t-size-s t-button--base t-button--default t-button--shape-rectangle"
          role="button"
        >
          <!---->
          <span
            class="t-button__text"
          >
            
            按钮
            
          </span>
        </button>
        
      </div>
    </div>
    <div
      class="badge-line"
    >
      <span>
        offset: [10, -10]
      </span>
      <div
        class="t-badge badge-item"
      >
        <div
          class="t-badge__inner t-badge--has-children t-badge--medium t-badge--circle"
          style="right: 10px; top: -10px;"
        >
          12
        </div>
        
        <button
          aria-disabled="false"
          class="t-button t-size-s t-button--base t-button--default t-button--shape-rectangle"
          role="button"
        >
          <!---->
          <span
            class="t-button__text"
          >
            
            按钮
            
          </span>
        </button>
        
      </div>
    </div>
  </div>
</div>
`;

exports[`Badge > Badge shapeVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
>
  <div
    class="badge-demo"
  >
    <div
      class="t-badge badge-item"
    >
      <div
        class="t-badge__inner t-badge--has-children t-badge--medium t-badge--circle"
      >
        12
      </div>
      
      <button
        aria-disabled="false"
        class="t-button t-size-s t-button--base t-button--default t-button--shape-rectangle"
        role="button"
      >
        <!---->
        <span
          class="t-button__text"
        >
          
          circle
          
        </span>
      </button>
      
    </div>
    <div
      class="t-badge badge-item"
    >
      <div
        class="t-badge__inner t-badge--has-children t-badge--medium t-badge--round"
      >
        12
      </div>
      
      <button
        aria-disabled="false"
        class="t-button t-size-s t-button--base t-button--default t-button--shape-rectangle"
        role="button"
      >
        <!---->
        <span
          class="t-button__text"
        >
          
          round
          
        </span>
      </button>
      
    </div>
    <div
      class="t-cell-group"
    >
      <div
        class="t-cell-group__title"
      >
        
        列表带徽标
        
      </div>
      <div
        class="t-cell-group__container"
      >
        
        <div
          class="t-cell t-cell--middle t-cell--bordered"
          label="单行标题"
        >
          <div
            class="t-cell__left-icon"
          >
            <!--v-if-->
            <!--v-if-->
          </div>
          <!--v-if-->
          <div
            class="t-cell__note"
          >
            
            <div
              class="t-badge"
            >
              <div
                class="t-badge__inner t-badge--medium t-badge--dot"
              >
                
              </div>
              <!---->
            </div>
            
          </div>
          <!--v-if-->
        </div>
        <div
          class="t-cell t-cell--middle t-cell--bordered"
          label="单行标题"
        >
          <div
            class="t-cell__left-icon"
          >
            <!--v-if-->
            <!--v-if-->
          </div>
          <!--v-if-->
          <div
            class="t-cell__note"
          >
            
            <div
              class="t-badge t-badge__ribbon--outer"
            >
              <!--v-if-->
              NEW
            </div>
            
          </div>
          <!--v-if-->
        </div>
        
      </div>
    </div>
  </div>
</div>
`;

exports[`Badge > Badge textVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
  data-v-9223ea9e=""
>
  <div
    class="badge-demo"
    data-v-9223ea9e=""
  >
    <div
      class="t-badge badge-item"
      data-v-9223ea9e=""
    >
      <!--v-if-->
      
      <button
        aria-disabled="false"
        class="t-button t-size-s t-button--base t-button--default t-button--shape-rectangle"
        data-v-9223ea9e=""
        role="button"
      >
        <!---->
        <span
          class="t-button__text"
        >
          
          NEW
          
        </span>
      </button>
      
    </div>
    <div
      class="t-badge badge-item"
      data-v-9223ea9e=""
    >
      <!--v-if-->
      
      <button
        aria-disabled="false"
        class="t-button t-size-s t-button--base t-button--default t-button--shape-rectangle"
        data-v-9223ea9e=""
        role="button"
      >
        <!---->
        <span
          class="t-button__text"
        >
          
          省略号
          
        </span>
      </button>
      
    </div>
  </div>
  <div
    class="badge-wrap"
    data-v-9223ea9e=""
  >
    <div
      class="t-cell-group"
      data-v-9223ea9e=""
    >
      <!--v-if-->
      <div
        class="t-cell-group__container"
      >
        
        <div
          class="t-cell t-cell--middle t-cell--bordered"
          data-v-9223ea9e=""
          value-align="left"
        >
          <div
            class="t-cell__left-icon"
          >
            <!--v-if-->
            <!--v-if-->
          </div>
          <!--v-if-->
          <div
            class="t-cell__note"
          >
            
            <span
              data-v-9223ea9e=""
            >
              单行标题
            </span>
            <div
              class="badge-tag-wrap"
              data-v-9223ea9e=""
            >
              <div
                class="t-badge badge-tag"
                data-v-9223ea9e=""
              >
                <!--v-if-->
                NEW
              </div>
              <div
                class="t-badge badge-tag"
                data-v-9223ea9e=""
              >
                <!--v-if-->
                NEW
              </div>
              <div
                class="t-badge badge-tag"
                data-v-9223ea9e=""
              >
                <!--v-if-->
                NEW
              </div>
              <div
                class="t-badge badge-tag"
                data-v-9223ea9e=""
              >
                <!--v-if-->
                NEW
              </div>
            </div>
            
          </div>
          <!--v-if-->
        </div>
        
      </div>
    </div>
  </div>
</div>
`;
